jobbergate-api/jobbergate_api/apps/job_scripts/routers.py

```diff
@@ -181,18 +165,32 @@ async def job_script_create(
     job_script_data_as_string = build_job_script_data_as_string(s3_application_tar, param_dict)
 
     sbatch_params = create_dict.pop("sbatch_params", [])
-    create_dict["job_script_data_as_string"] = inject_sbatch_params(job_script_data_as_string, sbatch_params)
+    job_script_data_as_string = inject_sbatch_params(job_script_data_as_string, sbatch_params)
 
     logger.debug("Inserting job_script")
-    try:
-        insert_query = job_scripts_table.insert().returning(job_scripts_table)
-        job_script_data = await database.fetch_one(query=insert_query, values=create_dict)
 
-    except INTEGRITY_CHECK_EXCEPTIONS as e:
-        raise HTTPException(status_code=status.HTTP_422_UNPROCESSABLE_ENTITY, detail=str(e))
+    async with database.transaction():
+
+        try:
+            insert_query = job_scripts_table.insert().returning(job_scripts_table)
+            job_script_data = await database.fetch_one(query=insert_query, values=create_dict)
+
+            assert job_script_data is not None
```

Comment:
shouldn't 400 be returned instead of AssertionError? This AssertionError could be hard to debug

Reply:
mypy...... such a boring thing sometimes

Reply:
Comment adressed at [5624142](https://github.com/omnivector-solutions/jobbergate/pull/111/commits/56241424c53fcd0e8abcbb7fd51ec1508c84ff8d).